databallpy/features/covered_distance.py

```diff
@@ -0,0 +1,473 @@
+                gradient_x = _filter_data(
+                    gradient_x,
+                    filter_type=filter_type,
+                    window_length=window,
+                    polyorder=poly_order,
+                )
+                gradient_y = _filter_data(
+                    gradient_y,
+                    filter_type=filter_type,
+                    window_length=window,
+                    polyorder=poly_order,
+                )
+
+            res_dict[column_id + f"_{new_name[0]}x"] = np.array(gradient_x)
+            res_dict[column_id + f"_{new_name[0]}y"] = np.array(gradient_y)
+            res_dict[column_id + f"_{new_name}"] = np.linalg.norm(
+                [gradient_x, gradient_y], axis=0
+            )
+
+        new_columns_df = pd.DataFrame(res_dict)
+
+        if inplace:
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", category=pd.errors.PerformanceWarning)
+                df[new_columns_df.columns] = new_columns_df
+            return None
+        else:
+            return pd.concat([df, new_columns_df], axis=1)
+
+    except Exception as e:
+        LOGGER.exception(f"Found unexpected exception in _differentiate: \n{e}")
+        raise e
+
+
+#%% 3-7-2024
+def _parse_intervals(intervals):
+    if not all(isinstance(element, (int, float)) for element in intervals):
+        raise TypeError("All elements in the tuple must be integers or floats")
+    
+    if len(intervals) % 2 != 0:
+        raise ValueError("Intervals must contain an even number of elements.")
+    
+    return [(min(intervals[i], intervals[i+1]), max(intervals[i], intervals[i+1])) for i in range(0, len(intervals), 2)]
+
+def get_covered_distance(
+    tracking_data: pd.DataFrame,
+    player_ids: list[str],
+    framerate: int,
+    velocity_intervals: tuple[float, ...] | None = None,
+    acceleration_intervals: tuple[float, ...] | None = None
+    ) -> dict:
+    """Function that calculates the distance covered in specified velocity and acceleration intervals
+
+    Args:
+        tracking_data:      tracking data
+        player_ids:         list containing players. Example = ['home_8', 'away_3', 'home_16']
+        framerate:          sample frequency in Hertz
+        velocity_intervals: tuple that contains the velocity interval(s). For example [0 3] & [4 8]: (0,3,8,4)
+        acc_intervals:      tuple that contains the acceleration interval(s). For example [-3 0] & [3 5]: (-3,0,3,5)      
+
+    Returns:
+        dict: a dictionary with for every player the total distance covered and optionally the 
+        distance covered within the given velocity and/or acceleration threshold(s)
+    
+    Notes: 
+        The function requires the velocity for every player calculated with the add_velocity function. 
+        The acceleration for every player depends on the presence of acceleration intervals in the input
+    """
+
+    try:
+        # Check input types
+        # tracking_data
```

Comment:
Try to get rid of uneccesarry commented code/information. It is very clear that here the input tracking data is checked, extra comments are not neccessary. In general, if you have clear code, good variable names and good documentation, inline comments are almost never necessary.